apps/meteor/server/models/raw/LivechatRooms.ts

```diff
@@ -1220,6 +1220,26 @@ export class LivechatRoomsRaw extends BaseRaw<IOmnichannelRoom> implements ILive
 		return this.col.aggregate(params, { readPreference: readSecondaryPreferred() });
 	}
 
+	findPaginatedRoomsByVisitorsIdsAndSource({
+		visitorsIds,
+		source,
+		options = {},
+	}: {
+		visitorsIds: string[];
+		source?: string;
+		options: FindOptions;
+	}): FindPaginated<FindCursor<IOmnichannelRoom>> {
+		return this.findPaginated<IOmnichannelRoom>(
+			{
+				'v._id': { $in: visitorsIds },
+				...(source && {
+					$or: [{ 'source.type': new RegExp(escapeRegExp(source), 'i') }, { 'source.alias': new RegExp(escapeRegExp(source), 'i') }],
+				}),
```

Comment:
Do we have a proper index for covering this query?

Reply:
We already have an index for `v_.id`, and since the `source` will not always be present, I believe that should be enough. Also, we already have more than 40 indexes for rooms. This could start affecting our writing performance, couldn't it?

Reply:
Yup, which makes me think: it's a requirement to search by type/alias? The query should work pretty normal in normal conditions, but if the number of visitors is big (since a contact apparently don't have a limit on how many visitors can be added to them) and visitors can have unlimited rooms, then this could be a problem. 🤔

Reply:
There's no limit but the number of visitors per contact is unlikely to ever be large. Even if a contact has a large number of rooms, this is a very specific query that can take a bit longer in that case without any issue. It's better than adding yet another index to the rooms collection for something that won't be relevant to most workspaces.

Also adding the filter by type/alias will just filter the rooms that have already been searched with an index, so it won't make it any slower.